AggregateMetrics/AggregateMetrics/AzureWebApp/DefaultCounters.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.AzureWebApp
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.ApplicationInsights.DataContracts;
+
+    /// <summary>
+    /// Class that gives the user the default performance counters.
+    /// </summary>
+    internal class DefaultCounters
+    {
+        private static readonly DefaultCounters DefaultCountersInstance = new DefaultCounters();
+
+        Dictionary<FlexiblePerformanceCounterGauge, MetricTelemetry> defaultCounters;
+        Dictionary<FlexiblePerformanceCounterGauge, RateCounter> rateDefaultCounters;
+
+        SumUpGauge processorTime = new SumUpGauge(
+                "processorTime",
+                new FlexiblePerformanceCounterGauge("kernelTime"),
+                new FlexiblePerformanceCounterGauge("userTime"));
+
+        SumUpGauge ioDataBytesRate = new SumUpGauge(
+                "ioDataBytesRate",
+                new FlexiblePerformanceCounterGauge("readIoBytes"),
+                new FlexiblePerformanceCounterGauge("writeIoBytes"),
+                new FlexiblePerformanceCounterGauge("otherIoBytes"));
+
+        private DefaultCounters()
+        {
+            this.Initialize();
+        }
+
+        /// <summary>
+        /// Gets the only instance of DefaultCounters.
+        /// </summary>
+        public static DefaultCounters Instance
+        {
+            get
+            {
+                return DefaultCountersInstance;
+            }
+        }
+
+        /// <summary>
+        /// Initializes the dictionaries for the default performance counters.
+        /// </summary>
+        public void Initialize()
+        {
+            this.defaultCounters = new Dictionary<FlexiblePerformanceCounterGauge, MetricTelemetry>()
+            {
+                { new FlexiblePerformanceCounterGauge("appRequestExecTime"), new MetricTelemetry() },
+                { new FlexiblePerformanceCounterGauge("privateBytes"), new MetricTelemetry() },
+                { new FlexiblePerformanceCounterGauge("requestsInApplicationQueue"), new MetricTelemetry() }
+            };
+
+            this.rateDefaultCounters = new Dictionary<FlexiblePerformanceCounterGauge, RateCounter>()
+            {
+                { new FlexiblePerformanceCounterGauge("requestsTotal"), new RateCounter() },
+                { new FlexiblePerformanceCounterGauge("exceptionsThrown"), new RateCounter() }
+            };
+        }
+
+        /// <summary>
+        /// Gets the default performance counters.
+        /// </summary>
+        /// <returns> List of MetricTelemetry objects</returns>
+        private List<MetricTelemetry> GetCounters()
```

Comment:
this code should not be here. Telemetry Module should take a list of gauges and register them. So GetValueAndReset will be called by aggregation module